internal/provider/protocols/protocols_s3_policy_resource.go

```diff
@@ -0,0 +1,439 @@
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/utils"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &ProtocolsS3PolicyResource{}
+var _ resource.ResourceWithImportState = &ProtocolsS3PolicyResource{}
+
+
+
+// NewProtocolsS3PolicyResource is a helper function to simplify the provider implementation.
+func NewProtocolsS3PolicyResource() resource.Resource {
+	return &ProtocolsS3PolicyResource{
+		config: connection.ResourceOrDataSourceConfig{
+			Name: "s3_policy",
+		},
+	}
+}
+
+// ProtocolsS3PolicyResource defines the resource implementation.
+type ProtocolsS3PolicyResource struct {
+	config connection.ResourceOrDataSourceConfig
+}
+
+// ProtocolsS3PolicyStatement describes the statement data model.
+type ProtocolsS3PolicyStatement struct {
+	SID        types.String `tfsdk:"sid"`
+	Effect     types.String `tfsdk:"effect"`
+	Actions    types.Set    `tfsdk:"actions"`
+	Resources  types.Set    `tfsdk:"resources"`
+	Index      types.Int64  `tfsdk:"index"`
+}
+
+// ProtocolsS3PolicyResourceModel describes the resource data model.
+type ProtocolsS3PolicyResourceModel struct {
+	CxProfileName types.String                      `tfsdk:"cx_profile_name"`
+	Name          types.String                      `tfsdk:"name"`
+	Comment       types.String                      `tfsdk:"comment"`
+	SVMName       types.String                      `tfsdk:"svm_name"`
+	ReadOnly      types.Bool                        `tfsdk:"read_only"`
+	Statements    []ProtocolsS3PolicyStatement      `tfsdk:"statements"`
+	ID            types.String                      `tfsdk:"id"`
+}
+
+// Metadata returns the resource type name.
+func (r *ProtocolsS3PolicyResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + r.config.Name
+}
+
+// Schema defines the schema for the resource.
+func (r *ProtocolsS3PolicyResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Protocols S3 Policy resource",
+
+		Attributes: map[string]schema.Attribute{
+			"cx_profile_name": schema.StringAttribute{
+				MarkdownDescription: "Connection profile name",
+				Required:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "S3 policy name",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"svm_name": schema.StringAttribute{
+				MarkdownDescription: "Name of the SVM",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"comment": schema.StringAttribute{
+				MarkdownDescription: "Optional comment for the S3 policy",
+				Optional:            true,
+			},
+			"read_only": schema.BoolAttribute{
```

Comment:
In the [document](https://docs.netapp.com/us-en/ontap-restapi-98/post-protocols-s3-services-policies.html#parameters), read_only should not be specified in the POST method. So this parameter type should be computed and not optional. And it should not have default value. Beware of the default value on the parameter. Usually the value will be provided on the response of api call. The default value should not be provided at the declaration.

				Computed: true,
				PlanModifiers: []planmodifier.Bool{
					boolplanmodifier.UseStateForUnknown(),
				},